xdf.cpp

```diff
@@ -176,251 +177,92 @@ int Xdf::load_xdf(std::string filename)
+                    if (tsBytes == 8)
                     {
-                        //read or deduce time stamp
-                        auto tsBytes = readBin<uint8_t>(file);
-
-                        double ts;  //temporary time stamp
-
-                        if (tsBytes == 8)
-                        {
-                            Xdf::readBin(file, &ts);
-                            streams[index].time_stamps.emplace_back(ts);
-                        }
-                        else
-                        {
-                            ts = streams[index].last_timestamp + streams[index].sampling_interval;
-                            streams[index].time_stamps.emplace_back(ts);
-                        }
-
-                        streams[index].last_timestamp = ts;
-
-                        //read the data
-                        for (int v = 0; v < streams[index].info.channel_count; ++v)
-                        {
-                            float data;
-                            Xdf::readBin(file, &data);
-                            streams[index].time_series[v].emplace_back(data);
-                        }
+                      Xdf::readBin(file, &ts);
+                      streams[index].time_stamps.emplace_back(ts);
                     }
-                }
-                else if (streams[index].info.channel_format.compare("double64") == 0)
-                {
-                    //if the time series is empty
-                    if (streams[index].time_series.empty())
-                        streams[index].time_series.resize(streams[index].info.channel_count);
-
-                    //for each sample
-                    for (size_t i = 0; i < numSamp; i++)
+                    else
                     {
-                        //read or deduce time stamp
-                        auto tsBytes = readBin<uint8_t>(file);
-
-                        double ts;  //temporary time stamp
-
-                        if (tsBytes == 8)
-                        {
-                            Xdf::readBin(file, &ts);
-                            streams[index].time_stamps.emplace_back(ts);
-                        }
-                        else
-                        {
-                            ts = streams[index].last_timestamp + streams[index].sampling_interval;
-                            streams[index].time_stamps.emplace_back(ts);
-                        }
-
-                        streams[index].last_timestamp = ts;
+                      ts = streams[index].last_timestamp + streams[index].sampling_interval;
+                      streams[index].time_stamps.emplace_back(ts);
+                    }
+                    
+                    streams[index].last_timestamp = ts;
+                    
+                    if(streams[index].info.channel_format.compare("string") == 0) {
+                      
+                      std::vector<std::variant<int, float, double, int64_t, std::string>> data;
```

Comment:
Is this declaration used? Seems it wasn't referenced anywhere.

Reply:
Agreed that line 208 is not needed.  It was part of an earlier attempt to parse the type of numeric variables stored in otherwise string data, which I ultimately abandoned.